### PR TITLE
Update nwaku submodules

### PR DESCRIPTION
**Pull request by jm-clius**
This PR updates all submodules for `nwaku`.
It closes https://github.com/status-im/nwaku/issues/1102 and closes https://github.com/status-im/nwaku/issues/1085

## Scope of changes

There were several breaking changes that had to be addressed. In order to ease reviewing, I've tried to address discrete issues in separate commits, but inevitably some follow-ups were needed.
Incomplete list of breaking changes that had to be addressed:
- ambiguity in `SIGTERM`
- move from `bearssl/decls` RNG to `bearssl/rand`
- REST API: request `ContentType` no longer a `string`
- nim-libp2p now requires all mounted protocols to be explicitly started. This needs to be true before mounting if the switch has already started.
- remove auto-mounting of `relay` on all nodes (https://github.com/status-im/nwaku/issues/1085)
- many breaking API changes in nim-eth

### CODE REVIEW

File: waku/whisper/whispernodes.nim [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const
```
Comment (rymnc) [inline waku/whisper/whispernodes.nim:1]: Slightly off-topic: Should we have a CI workflow that autogenerates this file weekly?

**Review by rymnc — APPROVED** [review]
LGTM, tests pass on my machine (test2, without rln)

File: waku/whisper/whispernodes.nim (hunk 1 above)
Comment (jm-clius) [reply waku/whisper/whispernodes.nim:1]: We try not to meddle to much in the "stable" (and unused) v1 code, but, more importantly, `fleets.status.im` where this comes from is not supposed to be used as a source of truth - it was meant to be a tool of convenience. For Waku v2 we encode lists like these in a proper, signed merkle tree structure hosted on a domain server somewhere.

**Review by LNSD — COMMENTED** [review]
Great and epic work! Congrats! 💪🏼 🥇 

I want to take some time to review and understand properly the "start protocol" changes. I don't see anything blocking so far. If you think you need to merge it right now, I am ok with it.

File: waku/v2/node/jsonrpc/jsonrpc_utils.nim [hunk 2]
```diff
@@ -54,7 +54,7 @@ proc toWakuMessage*(relayMessage: WakuRelayMessage, version: uint32): WakuMessag
               version: version,
               timestamp: t) 
 
-proc toWakuMessage*(relayMessage: WakuRelayMessage, version: uint32, rng: ref BrHmacDrbgContext, symkey: Option[SymKey], pubKey: Option[keys.PublicKey]): WakuMessage =
+proc toWakuMessage*(relayMessage: WakuRelayMessage, version: uint32, rng: ref keys.HmacDrbgContext, symkey: Option[SymKey], pubKey: Option[keys.PublicKey]): WakuMessage =
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply waku/v2/node/jsonrpc/jsonrpc_utils.nim]: Good point! Addressed in [b9f6454](https://github.com/status-im/nwaku/pull/1123/commits/b9f6454b224cb16295f557697c266cfeaf09c2a2). These _were_ two different implementations when I opened the PR, but `keys` now indeed uses the same `rand.HmacDrbgContext`. Missed this instance of using a confusing qualifier when I reverted changes I had made for when they were different. Also highlights again confusing imports, as `keys` now exports `bearssl/rand` 🤷. More refactoring for later! 😄

File: examples/v2/chat2.nim [hunk 3]
```diff
@@ -214,9 +214,10 @@ proc publish(c: Chat, line: string) =
   when PayloadV1:
     # Use Waku v1 payload encoding/encryption
     let
+      rng = keys.newRng()
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply examples/v2/chat2.nim:217]: `keys` does now use the `bearssl/rand` module's `HmacDrbgContext`. As for the `newRng()` utility we have an option between using the one in `libp2p/crypto` or `keys` (both now returns the same thing). `keys` seem to fit the v1 payload focus here a bit better, I think.

**Comment by jm-clius** [comment]
> I want to take some time to review and understand properly the "start protocol" changes.

@LNSD indeed! I think we have a scope for sensible refactoring here. In fact, I was hoping to move all "mount" logic from the node into either the protocols itself or a separate module that manages mounting logic. Became clear while addressing this PR though that this is too involved for a PR that's already way beyond bite-size :)

**Pull request merged**